tests/waveforms/cbc/test_cbc_waveforms.py

```diff
@@ -136,18 +247,83 @@ def test_taylor_f2(
+                atol=1e-3,
+            )
+            assert np.allclose(
+                1e21 * hc_lal_data.real,
+                1e21 * hc_ml4gw.real.numpy(),
+                atol=1e-3,
+            )
+            assert np.allclose(
+                1e21 * hc_lal_data.imag,
+                1e21 * hc_ml4gw.imag.numpy(),
+                atol=1e-3,
+            )
+
+        # If benchmarking calculate errors to record
+        # test_2 refers to differences between lal and ml4gw
+        # TaylorF2 methods when mass ratio is flipped
+        if request.config.getoption("--benchmark") != 0:
+            test_2_hp_real_abs_err = np.max(
+                np.abs(1e21 * hp_lal_data.real - 1e21 * hp_ml4gw.real.numpy())
+            )
+            test_2_hp_imag_abs_err = np.max(
+                np.abs(1e21 * hp_lal_data.imag - 1e21 * hp_ml4gw.imag.numpy())
+            )
+            test_2_hc_real_abs_err = np.max(
+                np.abs(1e21 * hc_lal_data.real - 1e21 * hc_ml4gw.real.numpy())
+            )
+            test_2_hc_imag_abs_err = np.max(
+                np.abs(1e21 * hc_lal_data.imag - 1e21 * hc_ml4gw.imag.numpy())
+            )
+
+        # If benchmark mode is on, gather all relavent data to store
+        if request.config.getoption("--benchmark") != 0:
+            data = {
+                "test_1_hp_real_abs_err": test_1_hp_real_abs_err,
+                "test_1_hp_imag_abs_err": test_1_hp_imag_abs_err,
+                "test_1_hc_real_abs_err": test_1_hc_real_abs_err,
+                "test_1_hc_imag_abs_err": test_1_hc_imag_abs_err,
+                "test_2_hp_real_abs_err": test_2_hp_real_abs_err,
+                "test_2_hp_imag_abs_err": test_2_hp_imag_abs_err,
+                "test_2_hc_real_abs_err": test_2_hc_real_abs_err,
+                "test_2_hc_imag_abs_err": test_2_hc_imag_abs_err,
+                "chirp_mass": chirp_mass[i],
+                "mass_ratio": mass_ratio[i],
+                "chi1": chi1[i],
+                "chi2": chi2[i],
+                "distance": distance[i],
+                "phase": phase[i],
+                "theta_jn": theta_jn[i],
+                "f_ref": f_ref,
+            }
+
+        # append data to batch and write if batch size exceeds set batch size
+        if request.config.getoption("--benchmark") != 0:
+            batch_data.append(data)
+            if len(batch_data) >= batch_size:
+                for batch in batch_data:
+                    write_benchmark_data(filename, batch)
+                batch_data = []
+
+                if get_file_size(filename) >= 0.05:
+                    filename = get_next_file_name(base_filename)
+                    logging.info(f"Switching to new file: {filename}")
+
+    # If there is any remaining data in the batch, write it to the file
+    if request.config.getoption("--benchmark") != 0:
+        if batch_data:
+            for batch in batch_data:
+                write_benchmark_data(filename, batch)
```

Comment:
After writing everything to file, it would be great to have a function to aggregate all the files and datasets into a single file